### PR TITLE
derive ‘size’ from ‘reduce’

### PR DESCRIPTION
**Pull request by davidchambers**
See sanctuary-js/sanctuary#414 for prior discussion.

### CODE REVIEW

**Comment by safareli** [comment]
@davidchambers Tuple es well as These might return unpredictable result. technically you can define size for betraversables too :d  (but we don't have bitraversable in FL)

File: test/index.js [hunk 1]
```diff
@@ -1026,6 +1026,24 @@ test('reduce', function() {
...
+  eq(Z.size(['foo', 'bar']), 2);
+  eq(Z.size(['foo', 'bar', 'baz']), 3);
+  eq(Z.size(Nil), 0);
+  eq(Z.size(Cons('foo', Nil)), 1);
+  eq(Z.size(Cons('foo', Cons('bar', Nil))), 2);
+  eq(Z.size(Cons('foo', Cons('bar', Cons('baz', Nil)))), 3);
+  eq(Z.size(Identity('quux')), 1);
+  eq(Z.size(Nothing), 0);
+  eq(Z.size(Just(0)), 1);
+  eq(Z.size(Tuple('abc', 123)), 1);
```
Comment (davidchambers) [inline test/index.js]: > Tuple […] might return unpredictable result.

I agree that this is surprising initially, but it seems correct. Haskell agrees:

```haskell
> length ("abc", 123)
1
```

What do you suggest, @safareli?

Comment (gabejohnson) [inline test/index.js]: It's not unpredictable _if_ the user is familiar with how `reduce` works for their data type. `Foldable` is in the signature after all.

Comment (safareli) [inline test/index.js]: > reduce works for their data type

Exactly!

**Pull request merged**